tests/Tokenizer/Transformer/FirstClassCallableTransformerTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Tokenizer\Transformer;
+
+use PhpCsFixer\Tests\Test\AbstractTransformerTestCase;
+use PhpCsFixer\Tokenizer\CT;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Tokenizer\Transformer\FirstClassCallableTransformer
+ */
+final class FirstClassCallableTransformerTest extends AbstractTransformerTestCase
+{
+    /**
+     * @dataProvider provideProcessCases
+     * @requires PHP 8.1
+     */
+    public function testProcess(string $source, array $expectedTokens): void
+    {
+        $this->doTest($source, $expectedTokens);
+    }
+
+    public function provideProcessCases(): \Generator
+    {
+        yield 'set' => [
+            '<?php
+strlen(...);
+$closure(...);
+$invokableObject(...);
+$obj->method(...);
+$obj->$methodStr(...);
+($obj->property)(...);
+Foo::method(...);
+$classStr::$methodStr(...);
+self::{$complex . $expression}(...);
+\'strlen\'(...);
+[$obj, \'method\'](...);
+[Foo::class, \'method\'](...) ?>',
```

Comment:
Does it support comments and whitespace in here too?

Reply:
good call, added more test cases to show it does